diff --git a/CMakeLists.txt b/CMakeLists.txt
index b08c26d..4dd5cb9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -100,7 +100,13 @@ endif()
 
 # We typically need newer versions libcudacxx than availabled in the toolkit.  pull down specific version here
 message(STATUS "Need libcuda++. Finding...")
-set(LIBCUDACXX_VERSION "2.1.0" CACHE STRING "Version of libcudacxx to use")
+
+# Work around -- sometimes matx ends up getting a LIBCUDACXX_VERSION string that is set, but empty.
+string(STRIP "${LIBCUDACXX_VERSION}" LIBCUDACXX_VERSION_STRIPPED)
+if(NOT DEFINED "${LIBCUDACXX_VERSION}" OR "${LIBCUDACXX_VERSION_STRIPPED}" STREQUAL "")
+  set(LIBCUDACXX_VERSION "2.1.0" CACHE STRING "Version of libcudacxx to use" FORCE)
+endif()
+
 include(cmake/FindLibcudacxx.cmake)
 target_include_directories(matx INTERFACE "$<BUILD_INTERFACE:${LIBCUDACXX_INCLUDE_DIR}>")    
 
